sql/server_component/mysql_thd_attributes_imp.cc

```diff
@@ -50,6 +50,14 @@ DEFINE_BOOL_METHOD(mysql_thd_attributes_imp::get,
         *((bool *)inout_pvalue) = t->is_server_upgrade_thread();
       } else if (!strcmp(name, "is_init_file_thread")) {
         *((bool *)inout_pvalue) = t->is_init_file_system_thread();
+      } else if (!strcmp(name, "m_db")) {
+        *((LEX_CSTRING *)inout_pvalue) = t->db();
```

Comment:
it doesn't work this way: LEX_CSTRING doesn't for example carry charset info. You're better off using mysql_string strings (this is the official charset compatible way to return data) or you need to convert to a known charset (utf-8) and copy it in a buffer.